bundles/nl.esi.comma.types/src/nl/esi/comma/types/utilities/TypeUtilities.xtend

```diff
@@ -323,4 +323,14 @@ class TypeUtilities {
             MapTypeConstructor: '''map<«type.type.name»,«type.valueType.typeName»>'''
         }
     }
+    
+    def static VectorTypeConstructor vectorOf(TypeObject element) {
+        val vtc = TypesFactory.eINSTANCE.createVectorTypeConstructor
+        vtc.dimensions.add(TypesFactory.eINSTANCE.createDimension)
+        vtc.type = asType(element).type
+        return vtc
+    }
```

Comment:
Using a dispatch method to also allow vectors of vector, and narrow the parameter type to a TypeDecl.

```suggestion

    def static dispatch VectorTypeConstructor vectorOf(TypeDecl td) {
        val vtc = TypesFactory.eINSTANCE.createVectorTypeConstructor
        vtc.dimensions += TypesFactory.eINSTANCE.createDimension
        vtc.type = EcoreUtil.copy(td)
        return vtc
    }

    def static dispatch VectorTypeConstructor vectorOf(VectorTypeDecl vtd) {
        val vtc = EcoreUtil.copy(vtd.constructor)
        vtc.dimensions += TypesFactory.eINSTANCE.createDimension
        return vtc
    }
```